vmdb/app/models/classification.rb

```diff
@@ -310,12 +310,16 @@ def find_entry_by_name(name, region_id = self.my_region_number)
     tag.nil? ? nil : self.class.find_by_tag_id(tag.id)
   end
 
-  def self.find_by_name(name, region_id = self.my_region_number, ns = DEFAULT_NAMESPACE)
+  def self.find_tag_by_name(name, region_id = self.my_region_number, ns = DEFAULT_NAMESPACE)
     if region_id.nil?
-      tag = Tag.find_by_name(Classification.name2tag(name, 0, ns))
+      Tag.find_by_name(Classification.name2tag(name, 0, ns))
     else
-      tag = Tag.in_region(region_id).find_by_name(Classification.name2tag(name, 0, ns))
+      Tag.in_region(region_id).find_by_name(Classification.name2tag(name, 0, ns))
     end
+  end
```

Comment:
Why is a class method on `Classification` returning `Tag` objects?  Shouldn't this method be on the `Tag` class?  It seems like `name` is actually a classification name, and we should have a `Tag.find_by_classification_name`.